### PR TITLE
added docker create and start events

### PR DESCRIPTION
**Pull request by cgtexmex**
The docker persona event monitor will now publish create & start
events.  This is needed in support of docker stats.

### CODE REVIEW

File: lib/apiservers/engine/backends/eventmonitor.go [hunk 1]
```diff
@@ -198,19 +198,28 @@ func (m *PortlayerEventMonitor) monitor() error {
...
 
 	vc := cache.ContainerCache().GetContainer(event.Ref)
 	if vc == nil {
 		log.Errorf("Portlayer event for container %s but not found in cache", event.Ref)
 	}
+	var dockerEvent string
+
+	switch event.Event {
+	case plevents.ContainerStopped:
+		dockerEvent = "die"
```
Comment (vburenin) [inline lib/apiservers/engine/backends/eventmonitor.go]: Are these events are used somewhere else? Should not they be defined as constants?

File: lib/apiservers/engine/backends/eventmonitor.go [hunk 2]
```diff
@@ -198,19 +198,28 @@ func (m *PortlayerEventMonitor) monitor() error {
...
+	case plevents.ContainerStarted:
+		dockerEvent = "start"
+	case plevents.ContainerCreated:
+		dockerEvent = "create"
```
Comment (sflxn) [inline lib/apiservers/engine/backends/eventmonitor.go]: Do we need this in addition to the "create" event we generate in container.go during ContainerCreate?

Comment (cgtexmex) [inline lib/apiservers/engine/backends/eventmonitor.go]: nope -- thanks for pointing that out...

File: lib/apiservers/engine/backends/eventmonitor.go (hunk 1 above)
Comment (cgtexmex) [inline lib/apiservers/engine/backends/eventmonitor.go]: I created constants for most of the events and updated where appropriate...

File: lib/apiservers/engine/backends/eventmonitor.go [hunk 3]
```diff
@@ -198,19 +210,26 @@ func (m *PortlayerEventMonitor) monitor() error {
...
 	if vc == nil {
 		log.Errorf("Portlayer event for container %s but not found in cache", event.Ref)
 	}
+	var dockerEvent string
+
+	switch event.Event {
+	case plevents.ContainerStopped:
+		dockerEvent = containerDieEvent
+	case plevents.ContainerStarted:
+		dockerEvent = containerStartEvent
```
Comment (sflxn) [inline lib/apiservers/engine/backends/eventmonitor.go]: So, I should have brought this up before.  If we were to follow Docker's pattern, their events are generally the result of a request fullfilled and not the operation completing.  In other words, with the exception of 1 or 2 events (e.g. die), all their events are published in the routers at the end of their related function.  So for start, that would get published at the end of a successful ContainerStart.  Also, not all events will have a corresponding event from the portlayer.  I think we should move this event to ContainerStart.

Comment (cgtexmex) [inline lib/apiservers/engine/backends/eventmonitor.go]: I don't really care what docker's pattern is... :-)  When I updated this I had that thought...meaning move to start..anyhoo...I'll do that..

**Pull request merged**